WindowsDevicePortalWrapper/UnitTestProject/WDPMockImplementations/RestDelete.cs

```diff
@@ -43,7 +43,7 @@ public partial class DevicePortal
                 {
                     if (!response.IsSuccessStatusCode)
                     {
-                        throw new DevicePortalException(response);
+                        throw await DevicePortalException.CreateAsync(response);
```

Comment:
You shouldn't have async code in a constructor, so I change the exception creation to be a static async creator (see further), and changed all uses of it to this instead.